### PR TITLE
Add simple example using NumPy to vectorized function evaluation

### PR DESCRIPTION
**Pull request by jakobj**
It seems we were missing an example of how one would use NumPy to vectorize function evaluations. This PR contains such an example as an extension of our standard example.

### CODE REVIEW

**Comment by jakobj** [comment]
ping @HenrikMettler

File: examples/example_evo_regression_numpy.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        Target function
+
+    Returns
+    -------
+    dict
+        Dictionary containing the history of the evolution
+    Individual
+        Individual with the highest fitness in the last generation
+    """
+    population_params = {"n_parents": 10, "seed": 818821}
```
Comment (HenrikMettler) [inline examples/example_evo_regression_numpy.py:120]: Is there a reason to not use the default parameterization? (Except for `genome_params['n_inputs] = 2` ?

**Review by HenrikMettler — CHANGES_REQUESTED** [review]
Left a question about parameterization. 
Also did we ever find a solution to the problem of ever increasing number of examples? (I guess implementing #312 would partially answer that ^^)

File: examples/example_evo_regression_numpy.py (hunk 1 above)
Comment (jakobj) [reply examples/example_evo_regression_numpy.py:120]: no reason except staying close to the "evo_regression" example. i'd like to keep it this way to really make minimal changes to the existing one, but i created a new issue that we should update the examples to use default parameters where possible.

**Comment by jakobj** [comment]
Thanks for the review!

So far we didn't find a good solution. But in general, I think the increase in increased accessibility provided by many examples may outweigh the cost of maintaining them. So far I think it's still manageable.

**Review by HenrikMettler — APPROVED** [review]
Good solution with the seperate Issue and I agree with respect to the No. of examples.

**Pull request merged**